src/peeker/src/args.rs

```diff
@@ -391,3 +397,49 @@ where
     let d = Duration::from_millis(Deserialize::deserialize(deser)?);
     Ok(Some(d))
 }
+
+lazy_static! {
+    static ref BASHLIKE_ENV_VAR_PATTERN: Regex =
+        Regex::new(r"(?P<declaration>\$\{(?P<var>[^:}]+)(:-(?P<default>[^}]+))?\})").unwrap();
+}
+
+/// replace instances of `${VAR:-default}` with a local environment variable if
+/// present or the default value.
+///
+/// # Panics
+/// - If environment variable does not exist and no default value is provided.
+fn substitute_config_env_vars(contents: &str) -> String {
+    let mut parsed_contents = contents.to_string();
+    for cap in BASHLIKE_ENV_VAR_PATTERN.captures_iter(contents) {
+        let val = match env::var(cap.name("var").unwrap().as_str()) {
+            Ok(val) => val,
+            Err(_) => cap.name("default").unwrap().as_str().to_string(),
+        };
```

Comment:
I think this would lead to slightly nicer error messages with `unwrap_or_else`:

```suggestion
        let var = cap.name("var").unwrap().as_str();
        let val = match env::var(var) {
            Ok(val) => val,
            Err(_) => cap
                .name("default")
                .unwrap_or_else(|| panic!("Env Var is not set and has no default: {}", var))
                .as_str()
                .to_string(),
        };
```